### PR TITLE
feat(annotator): 🚀 introduce circle annotator

### PR DESCRIPTION
**Pull request by onuralpszr**
# Description

We have box annotator and for support different shape I added circle annotator and split init section and create base annotator for reduce duplicate code.

List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

-   [X] New feature (non-breaking change which adds functionality)

### How has this change been tested, please provide a testcase or example of how you tested the change?

```python

from ultralytics import YOLO
import supervision as sv
import cv2

circle_annotator = sv.CircleAnnotator()

model = YOLO("yolov8n.pt") 
results = model("bus.jpg")
detections = sv.Detections.from_ultralytics(results[0])
frame = cv2.imread("bus.jpg")
labels = [f"{model.model.names[class_id]}"
          for _, _, _, class_id, tracker_id in detections]

annotated_frame = circle_annotator.annotate(
                scene=frame.copy(), 
                detections=detections, 
                labels=labels
            )

cv2.imshow("frame", annotated_frame)
cv2.waitKey(0)

```

### CODE REVIEW

**Comment by onuralpszr** [comment]
Preview in video (used tracking example)

https://github.com/roboflow/supervision/assets/1688848/ef8fb06b-3b59-46b5-a892-263da96ce167

**Comment by hardikdava** [comment]
@onuralpszr We already implemented `sv.EllipseAnnotator` but it is not released yet. Related branch : redesign_annotators.

**Comment by onuralpszr** [comment]
@hardikdava  Ellipse stay under the object. But  circle I made is around object like Box so they are different but I forgot about that PR.

**Comment by onuralpszr** [comment]
> Hi @onuralpszr and @hardikdava 👋🏻! Thanks a lot for your patience. I'm trying to catch up, but it will probably take a few more days before I'm fully in rhythm.
> 
> As for the PR, @onuralpszr, I guess you got motivated by [this](https://github.com/roboflow/supervision/issues/370) issue?

Yes

**Comment by onuralpszr** [comment]
> Hi @onuralpszr and @hardikdava 👋🏻! Thanks a lot for your patience. I'm trying to catch up, but it will probably take a few more days before I'm fully in rhythm.
> 
> As for the PR, @onuralpszr, I guess you got motivated by [this](https://github.com/roboflow/supervision/issues/370) issue?
> 
> @hardikdava is right; we have a separate PR with different annotators, but I think we need to add them one by one. Going for all of them at the same time is hard to review. So, I'd like to go forward with this PR. I'll do the review.

I also reviewed our new annotator system, if we want to merge our big annotator change PR, I can move this one to over there too.  I can also handle "triangle" part too

File: supervision/detection/annotate.py [hunk 1]
```diff
@@ -147,6 +133,128 @@ def annotate(
         return scene
 
 
+class CircleAnnotator(BaseAnnotator):
```
Comment (SkalskiP) [inline supervision/detection/annotate.py:136]: I think that `CircleAnnotator` should only draw circles.

In the next PR, we should add `TextAnnotator` responsible for drawing labels with background. `TextAnnotator` could be used in combination with `CircleAnnotator`, `TriangleAnnotator`, `EllipseAnnotator`, and so on.

@onuralpszr please drop the whole text drawing logic, and keep only the circles drawing.

File: supervision/detection/base_annotate.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from typing import Union
+
+from supervision.draw.color import Color, ColorPalette
+
+
+class BaseAnnotator:
```
Comment (SkalskiP) [inline supervision/detection/base_annotate.py:6]: I like the `BaseAnnotator` idea, but we should drop all text-related properties from the class definition. Not every annotator will draw text. 

I'm not even sure we need `color` and `thickness` is the right choice. I was thinking about making `BlurAnnotator`. As the name suggests, it is responsible for blurring detections. If  you think about it, it doesn't need `color` nor `thickness`.

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by SkalskiP** [comment]
> I also reviewed our new annotator system, if we want to merge our big annotator change PR, I can move this one to over there too. I can also handle "triangle" part too

No, please go one by one. One of the reasons I moved so slowly last time was the size of PR (and the amount of other things I need to do at Roboflow). Let's keep the big picture in mind (ideas from the old PR), but move step by step. 🙏🏻

**Comment by onuralpszr** [comment]
> > I also reviewed our new annotator system, if we want to merge our big annotator change PR, I can move this one to over there too. I can also handle "triangle" part too
> 
> No, please go one by one. One of the reasons I moved so slowly last time was the size of PR (and the amount of other things I need to do at Roboflow). Let's keep the big picture in mind (ideas from the old PR), but move step by step. 🙏🏻

I was going to seperate PR already just said "as a next step" so all good.

File: supervision/detection/annotate.py (hunk 1 above)
Comment (onuralpszr) [reply supervision/detection/annotate.py:136]: @SkalskiP  I will do changes but can we merge other PR first then I can do changes after rebase and give you clean results ? or I can add this codes to over there with your changes and also keep clean ?

Comment (SkalskiP) [reply supervision/detection/annotate.py:136]: By `other PR` you mean #170?

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by onuralpszr** [comment]
superseded pull request by #386

**Pull request closed without merging**